README.md

```diff
@@ -98,142 +113,140 @@ oc login -u developer -p developer https://localhost:8443
+Select among the available plans. Select the "Create Project" in the drop-down box.
 
-* console - manage
-* deviceX - send\_temperature, recv\_control/deviceX
-* sparkdriver - recv\_temperature, send\_max
-* thermostat - recv\_max, send\_control*
+Use the same value for the "name" field. The address space will be provisioned and may take a few
+minutes.
 
-You can edit the groups for each user by editing the user and clicking on the groups tab. This ensures that none of the components can access addresses they should not access.
+![Provision](images/provision2.png)
 
-#### Creating messaging addresses
+Skip binding at this point, we will perform the bind later.
+
+![Provision](images/provision3.png)
+
+If you go to your project, you should see the service provisioning in progress.
+
+![MyApp](images/myapp1.png)
+
+Once the provisioning is complete, we can login to the console and create the addresses we need for
+the workshop.
+
+### Creating messaging addresses
 
 In EnMasse, you have the concepts of address spaces and addresses.
 
 An address space is a group of addresses that can be accessed through a single connection (per
 protocol). This means that clients connected to the endpoints of an address space can send messages
 to or receive messages from any address it is authorized to send messages to or receive messages
 from within that address space. An address space can support multiple protocols, which is defined by
-the address space type. In this workshop, we only have 1 address space 'default' of type 'standard'.
+the address space type.
+
+Each messaging service provisioned in the service catalog creates a new address space. Conceptually,
+an address space may share messaging infrastructure with other address spaces.
 
 An address is part of an address space and represents a destination used for sending and receiving
 messages. An address has a type, which defines the semantics of sending messages to and receiving
-messages from that address.
+messages from that address. An address also has a plan, which determines the amount of resources
+provisioned to support the address.
 
 In the 'standard' address space, we have 4 types of addresses.
 
-   * **multicast** : 'direct' one-to-many using dispatch router
-   * **anycast** : 'direct' peer-2-peer using dispatch router
-   * **queue** : queue on broker
-   * **topic** : pub/sub on broker
+   * **multicast** : 'direct' one-to-many
+   * **anycast** : 'direct' peer-2-peer
+   * **queue** : queue
+   * **topic** : pub/sub
 
-##### Creating addresses for this workshop
+### Creating addresses for this workshop
 
-Go to the console, and locate the 'console' route. Click on the link to get to the EnMasse console.
+Login to the messaging console URL given by the provisioned service. You should be redirected to the
+authentication service. On the side of the login form, you can see a button named "OpenShift".
+Click on that to authenticate your user using your OpenShift credentials.
```

Comment:
I did that using "developer" and "developer" as username and password. A blank page showed me some permissions which I granted but then another page with the Keycloak logo said me : 

WE'RE SORRY ...
Unexpected error when authenticating with identity provider

Reply:
Ah, there is a step missing when setting up openshift. For the redirect to work, you need to set --public-hostname=<your host ip> . This is because the '127.0.0.1.nip.io'  suffix used for 'local development' cannot be used as redirection since keycloak trying to resolve '127.0.0.1.nip.ip' ending up trying to connect to itself will cause errors.

Reply:
So I have to re-deploy the cluster !! Thanks ! (you know ... I hate you ;))

Reply:
Are you using oc cluster up or minishift when seeing this btw? I would think with minishift it should perhaps work out of the box...

Reply:
I'm using `oc cluster up` so maybe that's the reason why I see this think :-)